### PR TITLE
[flink] MySql CDC can now deal with multiple changes in one ALTER TABLE statement

### PR DESCRIPTION
**Pull request by tsreaper**
### Purpose

Currently `MySqlDebeziumJsonEventParser` parses `ALTER TABLE` statement with a regular expression. This regular expression can only deal with a single change in the statement.

This PR further utilizes the information in debizum json format to support multiple changes in one ALTER TABLE statement.

### Tests

* MySqlSyncTableActionITCase#testMultipleSchemaEvolutions

### API and Format 

N/A

### Documentation

N/A

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/NewDataFieldListProcessFunction.java [hunk 1]
```diff
@@ -21,44 +21,95 @@
...
 /**
- * A {@link ProcessFunction} to handle {@link SchemaChange}.
+ * A {@link ProcessFunction} to handle schema changes. New schema is represented by a list of {@link
+ * DataField}s.
  *
  * <p>NOTE: To avoid concurrent schema changes, the parallelism of this {@link ProcessFunction} must
  * be 1.
  */
-public class SchemaChangeProcessFunction extends ProcessFunction<SchemaChange, Void> {
+public class NewDataFieldListProcessFunction extends ProcessFunction<List<DataField>, Void> {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/NewDataFieldListProcessFunction.java:50]: `UpdateDataField`? New let me think about only add columns...

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/NewDataFieldListProcessFunction.java [hunk 2]
```diff
@@ -21,44 +21,95 @@
...
+        RowType oldRowType = schemaManager.latest().get().logicalRowType();
+        Map<String, DataField> oldFields = new HashMap<>();
+        for (DataField oldField : oldRowType.getFields()) {
+            oldFields.put(oldField.name(), oldField);
+        }
+
+        List<SchemaChange> result = new ArrayList<>();
+        for (DataField newField : newDataFieldList) {
+            if (oldFields.containsKey(newField.name())) {
+                DataField oldField = oldFields.get(newField.name());
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/NewDataFieldListProcessFunction.java:80]: I think here we can have a util class, it is a generic requirement.

**Comment by JingsongLi** [comment]
Left two minor comments

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/NewDataFieldListProcessFunction.java (hunk 2 above)
Comment (tsreaper) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/cdc/NewDataFieldListProcessFunction.java:80]: No need. Let's make it a utility method when we actually need to reuse it.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**